sdmetrics/reports/single_table/_properties/column_pair_trends.py

```diff
@@ -184,7 +184,11 @@ def _get_columns_data_and_metric(
             data_real = real_data[[column_name_1, column_name_2]]
             data_synthetic = synthetic_data[[column_name_1, column_name_2]]
 
-        return data_real, data_synthetic, metric
+        metric_parameters = {}
+        if metric == ContingencySimilarity and min(len(data_real), len(data_synthetic)) > 50000:
```

Comment:
Should we check the min or the max here? I think the metric handles the case where the subsample is larger than the data size, so we could force the subsample only for the larger dataset.

Reply:
Good catch, thanks! Done in [8990f35](https://github.com/sdv-dev/SDMetrics/pull/723/commits/8990f3585d7b2ed8632adff38d8711dab3d31981)